compat/win32/fscache.c

```diff
@@ -145,20 +147,44 @@ static void fsentry_release(struct fsentry *fse)
 	InterlockedDecrement(&(fse->refcnt));
 }
 
+static int xwcstoutfn(char *utf, int utflen, const wchar_t *wcs, int wcslen)
+{
+	if (!wcs || !utf || utflen < 1) {
+		errno = EINVAL;
+		return -1;
+	}
+	utflen = WideCharToMultiByte(CP_UTF8, 0, wcs, wcslen, utf, utflen, NULL, NULL);
```

Comment:
Shouldn't  we handle the case where wcslen is zero?

Reply:
I believe we already do.  A filename with zero length will be an error.  Since we pass wcslen through to WideCharToMultiByte it will return an error which gets passed out to the caller.

Reply:
I meant this routine may not want to fail converting zero length string, but if that's intentional because this will be used only for converting file names, that makes sense.